crest/Assets/Crest/Crest/Scripts/Collision/SamplingHelpers.cs

```diff
@@ -15,16 +15,30 @@ public class SampleHeightHelper
 
         float _minLength = 0f;
 
+#if UNITY_EDITOR
+        int _lastFrame = -1;
+#endif
+
         /// <summary>
-        /// Call this to prime the sampling
+        /// Call this to prime the sampling. The SampleHeightHelper is good for one query per frame - if it is called multiple times in one frame
+        /// it will throw a warning. Calls from FixedUpdate are an exception to this - pass true as the last argument to disable the warning.
         /// </summary>
         /// <param name="i_queryPos">World space position to sample</param>
         /// <param name="i_minLength">The smallest length scale you are interested in. If you are sampling data for boat physics,
         /// pass in the boats width. Larger objects will ignore small wavelengths.</param>
-        public void Init(Vector3 i_queryPos, float i_minLength)
+        /// <param name="fromFixedUpdate">Pass true if calling from FixedUpdate(). This will omit a warning when there on multipled-FixedUpdate frames.</param>
+        public void Init(Vector3 i_queryPos, float i_minLength = 0f, bool fromFixedUpdate = false)
         {
             _queryPos[0] = i_queryPos;
             _minLength = i_minLength;
+
+#if UNITY_EDITOR
+            if (_lastFrame >= Time.frameCount && !fromFixedUpdate)
```

Comment:
This is minor but it might be more optimal to shortcut on the boolean check?

Reply:
What do you mean - do you mean switch the order so the !fromFixedUpdate is evaluated first?

Reply:
Sorry, yes.

Reply:
Done